cookbook/core/type_system/flyte_pickle.py

```diff
@@ -0,0 +1,52 @@
+"""
+.. _flyte_pickle:
+
+Using Flyte Pickle
+----------------------------
+
+Flyte enforces type safety by leveraging type information to be able to compile
+tasks/workflows, which enables all sorts of nice features (like static analysis of tasks/workflows, conditional branching, etc.)
+
+However, we do also want to provide enough flexibility to end-users so that they don't have to put in a lot of up front
+investment figuring out all the types of their data structures before experiencing the value that flyte has to offer.
+
+Flyte supports FlytePickle transformer which will convert any unrecognized type in type hint to
+FlytePickle, and serialize / deserialize the python value to / from a pickle file.
+
+This example shows how users can custom object without register a transformer.
+"""
+from flytekit import task, workflow
+
+
+# %%
+# This People is a user defined complex type, which can be used to pass complex data between tasks.
+# We will serialize this class to a pickle file and pass it between different tasks.
+#
```

Comment:
Also point to data class example as a preferred solution for this example

Reply:
done, thanks.